third_party/txt/src/minikin/Layout.cpp

```diff
@@ -284,10 +286,20 @@ hb_font_funcs_t* getHbFontFuncs(bool forColorBitmapFont) {
   return *funcs;
 }
 
+// Stores the result of isColorBitmapFont to ensure lag-free emoji rendering
+static std::unordered_map<hb_face_t*, bool> sIsColorBitmapFontCache;
```

Comment:
Using an hb_face_t* as the cache key is unsafe because Layout doesn't control the lifetime of the hb_face_t.

The isColorBitmapFont cache entry needs to be deleted if the font/face is destroyed.

Reply:
You're right, I found a better solution using a HarfBuzz method that directly checks for the existence of a CBDT table (https://harfbuzz.github.io/harfbuzz-hb-ot-color.html#hb-ot-color-has-png).

I will close this PR and submit a new one.

Reply:
Here is the new PR: https://github.com/flutter/engine/pull/7686